crates/rpc/src/lib.rs

```diff
@@ -42,9 +42,9 @@ impl RpcServer {
     }
 
     /// Starts the HTTP-RPC server.
-    pub async fn run(self) -> Result<(HttpServerHandle, SocketAddr), anyhow::Error> {
-        let server = HttpServerBuilder::default()
-            .set_middleware(self.middleware)
+    pub async fn run(self) -> Result<(ServerHandle, SocketAddr), anyhow::Error> {
+        let server = ServerBuilder::default()
+            .set_logger(self.middleware)
```

Comment:
Should we rename `self.middleware` to `self.logger`.

Reply:
Or is it actually our `metrics` middleware?

Reply:
Looks like in the meantime `jsonrpsee` switched from _middleware_ to _logger_ to describe the functionality of adding callbacks that allow you to take notice of certain events. This happened probably due to the fact that at that time they needed to differentiate from _middleware_ understood as `tower` _middleware_ which `jsonrpsee` started supporting.

So, yes, it is our `metrics` middleware. I could rename it all to logger (`self.middleware` to `self.logger`, `metrics::middleware::RpcMetricsMiddleware` to `metrics::logger::RpcMetricsLogger`, and so on). WDYT?

Reply:
Hmm honestly your call - whatever you think makes sense. I guess the type signature will help no matter what.